### DISCUSSION TITLE
Graphical artefacts in 3D scene using Mesa3D on Windows

### DISCUSSION THREAD

**Discussion by DavidOsicka** [pal1000/mesa-dist-win#177] [Q&A]
Hello guys,

I have kind of old PC game. I don´t have a source code from the software anymore.
It renders a 3D scene and displays some overlays with text information on top of the scene.
The game was developed for Windows XP and OpenGL 2.0.
I am porting it to Windows 10 IoT Enterprise now and running it on Intel UHD 630 graphics with support for OpenGL 3.0 and above.
After lot of struggling and searching for different options I´ve found a way how to run the 3D map software on Windows 10 IoT using Mesa3D libraries - per application installation.
But there are some graphical artefacts being rendered on the scene (pictures attached) and the application is not stable and crashes sometimes.

Any ideas on how to solve it appreciated.
![error_1](https://github.com/pal1000/mesa-dist-win/assets/15103247/7ae022d1-49ca-46f4-b83f-2262c478ef3d)
![error_2](https://github.com/pal1000/mesa-dist-win/assets/15103247/92f07a08-8d57-4a83-9fa1-5ec82b92521e)
![error_3](https://github.com/pal1000/mesa-dist-win/assets/15103247/3306964b-c274-4fe6-8f17-5f2fd6a656e7)
![error_8](https://github.com/pal1000/mesa-dist-win/assets/15103247/2c9a02ae-1d9f-4ec1-a439-ee2849ce236a)
![error_9](https://github.com/pal1000/mesa-dist-win/assets/15103247/f52f5092-0ad9-476b-b127-e7980fcf0e6a)

**Comment by pal1000** [comment]
For graphics glitches you may need to collect OpenGL API call dumps since it's a software you are working on.
For crashes please collect call stack (MSVC) or backtrace (MinGW). Debug symbols are provided to help debugging.
Once you have useful logs please report [upstream](https://gitlab.freedesktop.org/mesa/mesa/-/issues/).

**Reply by DavidOsicka** [reply]
Dump log recorded by Apitrace tool is saved here: https://file.io/DdWMmimc1uIy

The log is recorded without using Mesa 3D. I think Apitrace uses the OpenGL that comes together with the tool.
Even if the Mesa3D libraries were in the same path as the exacutable .exe file it was not used when running in Apitrace.
But when I run it without Apitrace the Mesa3D libraries (.dll files) are used.
OS is Windows 10 IoT.

There is a lot of errors using the OpenGL and the image looks much more distorted, much more graphical artifacts.
When I use Mesa3D libraries (dxil.dll, libgalium_wgl.dll, libglapi.dll, opengl32.dll) the image is usually all right only sometimes the artifacts you see above appears there.
